rai/node/node.cpp

```diff
@@ -1479,17 +1489,61 @@ bool rai::block_processor::should_log ()
 bool rai::block_processor::have_blocks ()
 {
 	assert (!mutex.try_lock ());
-	return !blocks.empty () || !forced.empty ();
+	return !blocks.empty () || !forced.empty () || !state_blocks.empty ();
+}
+
+void rai::block_processor::verify_state_blocks (std::unique_lock<std::mutex> & lock_a)
+{
+	lock_a.lock ();
+	std::deque<std::pair<std::shared_ptr<rai::block>, std::chrono::steady_clock::time_point>> items;
+	items.swap (state_blocks);
+	lock_a.unlock ();
+	auto size (items.size ());
```

Comment:
`items.size ()` needs to be behind the lock I think, otherwise it's undefined behavior.

Reply:
Items is the local variable that's been swapped for the shared container, it should be safe.

Reply:
Yep, read it as the member variable :(